obspy/io/csv/core.py

```diff
@@ -0,0 +1,484 @@
+time or year, mon, day, hour, minu, sec
+lat, lon, dep, mag, magtype, id
+(see also global FIELDS variable and help of read_csv)
+
+Note: This plugin can be easily extended to write and read more event
+information.
+If you are interested, please send a PR to the github repository.
+  1. Add 'extended' or similar key to FIELDS dict, e.g. as a start use
+      'extended': (
+          '{time!s:.25} {lat:.6f} {lat_err:.6f} {lon:.6f} {lon_err:.6f} '
+          '{dep:.3f} {dep_err:.3f} {mag:.2f} {mag_err:.2f} {magtype} {id}'
+          )
+     You can also test by just passing the string to `fields` option.
+  2. Implement writing functionality in write_csv by adding the new properties
+     to the dict d. Missing values have to be handled.
+  3. Implement reading functionality for the new properties in read_csv.
+  4. Write some tests testing all new properties. Check that an event with
+     all new properties defined or missing can be written and read again.
+Similar can be done for the picks using PFIELDS, _write_picks, _read_picks
+"""
+
+import csv
+from contextlib import contextmanager
+import io
+import math
+from string import Formatter
+from warnings import warn
+import zipfile
+
+
+import numpy as np
+from obspy import UTCDateTime as UTC
+from obspy.core import event as evmod
+
+
+__version__ = '0.8.1-dev'
+
+# for reading
+DEFAULT = {'magtype': ''}
+# for writing
+FIELDS = {
+    'basic': (
+        '{id} {time!s:.25} {lat:.6f} {lon:.6f} {dep:.3f} '
+        '{magtype} {mag:.2f}'),
+    'eventtxt': (
+        '{id},{time!s:.25},{lat:.6f},{lon:.6f},{dep:.3f},{author},,{contrib},,'
+        '{magtype},{mag:.2f},{magauthor},{region}').split(',')
+}
+PFIELDS = {
+    'basic': '{seedid} {phase} {time:.5f} {weight:.3f}'
+}
+CSZ_COMMENT = f'CSZ format v{__version__} obspy_no_uncompress'.encode('utf-8')
+# for load_csv
+DTYPE = {
+    'time': 'datetime64[ms]',
+    'lat': float,
+    'lon': float,
+    'dep': float,
+    'mag': float,
+    'magtype': 'U10',
+    'id': 'U50'
+}
+
+# catalog and contribid are not used
+EVENTTXT_NAMES = (  # for reading
+    'id time lat lon dep author catalog contrib contribid '
+    'magtype mag magauthor region')
+EVENTTXT_HEADER = (  # for writing
+    '#EventID | Time | Latitude | Longitude | Depth/km | Author | Catalog | '
+    'Contributor | ContributorID | MagType | Magnitude | MagAuthor | '
+    'EventLocationName')
+
+
+def _is_csv(fname, **kwargs):
+    try:
+        return _read_csv(fname, format_check=True)
+    except Exception:
+        return False
+
+
+def _is_eventtxt(fname, **kwargs):
+    try:
+        return _read_eventtxt(fname, format_check=True)
+    except Exception:
+        return False
+
+
+def _is_csz(fname, **kwargs):
+    try:
+        assert zipfile.is_zipfile(fname)
+        with zipfile.ZipFile(fname) as zipf:
+            assert (zipf.comment.startswith(b'CSZ') and
+                    zipf.comment.endswith(b'obspy_no_uncompress'))
```

Comment:
If python is run in optimized mode (-O flag) this wont work because asserts are ignored.